crates/polars-plan/src/plans/optimizer/predicate_pushdown/join.rs

```diff
@@ -66,22 +66,27 @@ fn should_block_join_specific(
     }
 }
 
+/// Returns a tuple indicating whether predicates should be blocked for either side based on the
+/// join type.
+///
+/// * `true` indicates that predicates must not be pushed to that side
 fn join_produces_null(how: &JoinType) -> LeftRight<bool> {
-    #[cfg(feature = "asof_join")]
-    {
-        match how {
-            JoinType::Left => LeftRight(false, true),
-            JoinType::Full { .. } | JoinType::Cross | JoinType::AsOf(_) => LeftRight(true, true),
-            _ => LeftRight(false, false),
-        }
-    }
-    #[cfg(not(feature = "asof_join"))]
-    {
-        match how {
-            JoinType::Left => LeftRight(false, true),
-            JoinType::Full { .. } | JoinType::Cross => LeftRight(true, true),
-            _ => LeftRight(false, false),
```

Comment:
I think we forgot to update this function when we added `Right` join, and it silently passed because of the `_ =>` catch-all

Reply:
Don't silently catch all. :|